src/main/java/com/app/onlinebookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,40 @@
+package com.app.onlinebookstore.repository.book;
+
+import com.app.onlinebookstore.dto.BookSearchParameters;
+import com.app.onlinebookstore.model.Book;
+import com.app.onlinebookstore.repository.SpecificationBuilder;
+import com.app.onlinebookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book, BookSearchParameters> {
+    private static final String AUTHOR_KEY = "author";
+    private static final String TITLE_KEY = "title";
+    private static final String ISBN_KEY = "isbn";
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> specification = Specification.where(null);
+        if (searchParameters.author() != null && searchParameters.author().length > 0) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationParameter(AUTHOR_KEY)
+                    .getSpecification(searchParameters.author()));
+        }
+        if (searchParameters.isbn() != null && searchParameters.isbn().length > 0) {
+            specification = specification.and(bookSpecificationProviderManager
```

Comment:
Blocks of code that add parameter to specification are very similar, maybe it is possible to extract common code to method?